lib/srv/regular/sshserver.go

```diff
@@ -1264,20 +1264,22 @@ func (s *Server) HandleNewConn(ctx context.Context, ccx *sshutils.ConnectionCont
 	}
 
 	// Create host user.
-	created, userCloser, err := s.termHandlers.SessionRegistry.TryCreateHostUser(identityContext)
+	created, userCloser, err := s.termHandlers.SessionRegistry.UpsertHostUser(identityContext)
 	if err != nil {
-		return ctx, trace.Wrap(err)
+		log.Warnf("error while creating host users: %s", err)
```

Comment:
I'm not sure what this PR is really getting us now, weren't we already logging warnings? But the user rarely sees those logs. If we are going to close the linked issue I would love to see in the PR description what the user actually sees when encountering the error now and how we've improved it

I'm also against logging at the warning level for conditions that we totally expect, like not creating a host user for a teleport user that has no permissions to create host users, which is the expected path for most SSH connections probably

I wonder if we could instead update `(*Server).replyError` to try to detect errors related to the user not existing and add context to the error we send back to the user there

Reply:
Oh I just saw we're not actually trying to close https://github.com/gravitational/teleport/issues/40370 with this anymore

Reply:
Yeah part of this solution ended up overlapping with a bugfix for the second issue now linked in the description. I plan on opening another PR with the work to propagate errors appropriately to the client. For now I can at least reduce the log level, because I agree we shouldn't be warning for something expected out of a healthy system  :+1: